libs/remix-core-plugin/src/lib/openzeppelin-proxy.ts

```diff
@@ -1,26 +1,80 @@
 import { Plugin } from '@remixproject/engine';
-import { ContractData } from '../types/contract';
+import { ContractABI, ContractAST } from '../types/contract';
+import { UUPS, UUPSABI, UUPSBytecode, UUPSfunAbi } from './constants/uups';
 
 const proxyProfile = {
   name: 'openzeppelin-proxy',
   displayName: 'openzeppelin-proxy',
   description: 'openzeppelin-proxy',
-  methods: ['isConcerned', 'execute']
+  methods: ['isConcerned', 'execute', 'getDeployOptions']
 };
-
 export class OpenZeppelinProxy extends Plugin {
   blockchain: any
+  kind: 'UUPS' | 'Transparent'
   constructor(blockchain) {
     super(proxyProfile)
     this.blockchain = blockchain
   }
 
-  async isConcerned(contractData: ContractData): Promise<boolean> {
+  async isConcerned(ast: ContractAST = {} as ContractAST) {
     // check in the AST if it's an upgradable contract
+    if (ast.nodes && ast.nodes.find(node => node.absolutePath && node.absolutePath.includes(UUPS))) {
+      this.kind = 'UUPS'
+      return true
+    }
+    //
+    // else if transparent contract run check true/false
+    //
     return false
   }
 
-  async execute(contractData: ContractData, contractMetadata: any, compiledContracts: any) {
+  async getDeployOptions (contracts: ContractABI) {
+    if (this.kind === 'UUPS') {
+      const inputs = {}
+    
+      Object.keys(contracts).map(name => {
+        const abi = contracts[name].abi
+        const initializeInput = abi.find(node => node.name === 'initialize')
```

Comment:
how are you sure you are targeting the righ contract? should `getDeployOptions` have a parameter like `contractName: string` ?